Assets/VRTK/Scripts/VRTK_MoveInPlace.cs

```diff
@@ -0,0 +1,387 @@
+        /// <param name="HeadsetOnly">Track only headset for movement caluclations.</param>
+        public enum ControlOptions
+        {
+            HeadsetAndControllers,
+            ControllersOnly,
+            HeadsetOnly,
+        }
+
+        /// <summary>
+        /// Options for which method is used to determine player direction while moving.
+        /// </summary>
+        /// <param name="Gaze">Player will always move in the direction they are currently looking.</param>
+        /// <param name="DumbDecoupling">Player will move in the direction they were first looking when they engaged Move In Place.</param>
+        /// <param name="SmartDecoupling">Player will move in the direction they are looking only if their headset point the same direction as their controllers.</param>
+        public enum DirectionalMethod
+        {
+            Gaze,
+            DumbDecoupling,
+            SmartDecoupling
+        }
+
+        /// <summary>
+        /// If true, the left controller's trackpad will engage Move In Place.
+        /// </summary>
+        public bool LeftController
+        {
+            get { return leftController; }
+            set
+            {
+                leftController = value;
+                SetControllerListeners(controllerLeftHand);
+            }
+        }
+
+        /// <summary>
+        /// If true, the right controller's trackpad will engage Move In Place.
+        /// </summary>
+        public bool RightController
+        {
+            get { return rightController; }
+            set
+            {
+                rightController = value;
+                SetControllerListeners(controllerRightHand);
+            }
+        }
+
+        [Tooltip("If this is checked then the left controller touchpad will be enabled to move the play area. It can also be toggled at runtime.")]
+        [SerializeField]
+        private bool leftController = true;
+        [Tooltip("If this is checked then the right controller touchpad will be enabled to move the play area. It can also be toggled at runtime.")]
+        [SerializeField]
+        private bool rightController = true;
+
+        [Tooltip("Select which trackables are used to determine movement.")]
+        [SerializeField]
+        public ControlOptions controlOptions = ControlOptions.HeadsetAndControllers;
+
+        [SerializeField]
+        [Tooltip("Lower to decrease speed, raise to increase.")]
+        public float speedScale = 1;
+
+        [SerializeField]
+        [Tooltip("The max speed you can move in game units. (If 0 or less, max speed is uncapped)")]
+        public float maxSpeed = 4;
+
+        [SerializeField]
+        [Tooltip("How your movement direction will be determined.  The Gaze method tends to lead to the least motion sickness.  Smart decoupling is still a Work In Progress.")]
+        public DirectionalMethod dirMethod = DirectionalMethod.Gaze;
```

Comment:
rename this to `directionMethod` it's better to be verbose rather than shortening words.